CMake Error while running in RN 0.75 (newArchEnabled = true)

### Description

` CMake Error at /Users/kishore/Documents/NEWARCH/MyApp/node_modules/react-native/ReactAndroid/cmake-utils/ReactNative-application.cmake:138 (target_link_libraries):
    Cannot specify link libraries for target "react_codegen_RNTestlibSpec"
    which is not built by this project.
  Call Stack (most recent call first):
    CMakeLists.txt:31 (include) : com.android.ide.common.process.ProcessException: -- The C compiler identification is Clang 17.0.2
  -- The CXX compiler identification is Clang 17.0.2
  -- Detecting C compiler ABI info
  -- Detecting C compiler ABI info - done
  -- Check for working C compiler: /Users/kishore/Library/Android/sdk/ndk/26.1.10909125/toolchains/llvm/prebuilt/darwin-x86_64/bin/clang - skipped
  -- Detecting C compile features
  -- Detecting C compile features - done
  -- Detecting CXX compiler ABI info
  -- Detecting CXX compiler ABI info - done
  -- Check for working CXX compiler: /Users/kishore/Library/Android/sdk/ndk/26.1.10909125/toolchains/llvm/prebuilt/darwin-x86_64/bin/clang++ - skipped
  -- Detecting CXX compile features
  -- Detecting CXX compile features - done
  -- Configuring incomplete, errors occurred!
  See also "/Users/kishore/Documents/NEWARCH/MyApp/android/app/.cxx/Debug/3vw6c6w4/arm64-v8a/CMakeFiles/CMakeOutput.log".`


### Packages

- [X] create-react-native-library
- [X] react-native-builder-bob

### Selected options

<img width="380" alt="Screenshot 2024-10-23 at 6 48 19 PM" src="https://github.com/user-attachments/assets/929cfba1-4232-4098-b356-c641f592e124">
<img width="508" alt="Screenshot 2024-10-23 at 6 48 26 PM" src="https://github.com/user-attachments/assets/5441dc3a-87c3-47c8-a9e1-b072c5c2e242">
<img width="538" alt="Screenshot 2024-10-23 at 6 48 33 PM" src="https://github.com/user-attachments/assets/1e99f329-1514-4774-8376-68b9ca00e807">


### Link to repro

Have created new react native library with above options

### Environment

React Native

--- a/packages/create-react-native-library/templates/common-local/$package.json
+++ b/packages/create-react-native-library/templates/common-local/$package.json
@@ -6,7 +6,7 @@
 <% if (project.arch !== 'legacy') { -%>
   "codegenConfig": {
     "name": "RN<%- project.name -%><%- project.view ? 'View': '' -%>Spec",
-    "type": <%- project.view ? '"components"': '"modules"' %>,
+    "type": <%- project.view ? '"all"': '"modules"' %>,
     "jsSrcsDir": "src"
   },
 <% } -%>

--- a/packages/create-react-native-library/templates/common/$package.json
+++ b/packages/create-react-native-library/templates/common/$package.json
@@ -198,7 +198,7 @@
   },
   "codegenConfig": {
     "name": "RN<%- project.name -%><%- project.view ? 'View': '' -%>Spec",
-    "type": "<%- project.view ? 'components': 'modules' -%>",
+    "type": "<%- project.view ? 'all': 'modules' -%>",
     "jsSrcsDir": "src",
     "outputDir": {
       "ios": "ios/generated",
@@ -209,7 +209,7 @@
     <% if (example === 'vanilla') { -%>
     },
     "includesGeneratedCode": true
-    <% } else { -%>    
+    <% } else { -%>
     }
     <% } -%>
 <% } -%>

